### PR TITLE
Switch to Artifact Registry image

### PR DESCRIPTION
**Pull request by colinking**
This PR switches our EC2 AMI to use the agent image hosted on GCP Artifact Registry. This avoids issues with anonymous pull rate limiting from DockerHub.

Once merged, I will publish a patch release (`0.4.13`). Before merging, I'll rebuild the AMIs and update the AMI IDs.

cc @joshma

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #16** <a href="https://app.graphite.dev/github/pr/airplanedev/terraform-aws-airplane-cluster/16" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Comment by colinking** [comment]
cc @joshma having trouble with the build script:

```
...
==> Wait completed after 3 minutes 40 seconds

==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs: AMIs were created:
us-west-2: ami-03f249339e8b14a30

--> amazon-ebs: AMIs were created:
us-west-2: ami-03f249339e8b14a30


An error occurred (InvalidAMIID.NotFound) when calling the DescribeImages operation: The image id '[ami-03f249339e8b14a30]' does not exist
```

Have you seen this before?

File: packer/conf/bin/configure-airplane.sh [hunk 1]
```diff
@@ -12,6 +12,7 @@ AP_API_TOKEN=${AP_API_TOKEN}
 AP_TEAM_ID=${AP_TEAM_ID}
 AP_LABELS=cloud:aws aws_instance_id:${INSTANCE_ID} aws_region:${AWS_REGION} aws_account_id:${AWS_ACCOUNT_ID} ${AP_LABELS:-}
 AP_AUTO_UPGRADE=${AP_AUTO_UPGRADE}
+AP_AGENT_IMAGE=us-docker.pkg.dev/airplane-prod/public/agent:1
```
Comment (joshma) [inline packer/conf/bin/configure-airplane.sh:15]: (same comment, is this needed?)

**Comment by joshma** [comment]
> Have you seen this before?

I wonder if it's just a race condition? This seems like it exists: https://us-west-2.console.aws.amazon.com/ec2/v2/home?region=us-west-2#Images:visibility=owned-by-me;v=3;search=:ami-03f249339e8b14a30

File: packer/conf/bin/configure-airplane.sh (hunk 1 above)
Comment (colinking) [reply packer/conf/bin/configure-airplane.sh:15]: https://github.com/airplanedev/terraform-aws-airplane-agents/pull/7#discussion_r837684120

**Review by ericliuche — APPROVED** [review]

File: packer/build.sh [hunk 2]
```diff
@@ -4,6 +4,8 @@ set -euo pipefail
 # Fixes https://github.com/aws/aws-cli/pull/4702#issue-344978525
 export AWS_PAGER=""
 
+export AWS_REGION="us-west-2"
```
Comment (colinking) [inline packer/build.sh:7]: This fixes the build issue I hit. `us-west-2` was not my local config's default region.

File: main.tf [hunk 3]
```diff
@@ -6,11 +6,11 @@ data "aws_subnet" "subnet" {
...
-    "ap-southeast-1" = "ami-0a478b8aac8e6676f"
-    "ap-southeast-2" = "ami-007fc7a8cfc847277"
-    "eu-west-1"      = "ami-08072b9515a0afd47"
-    "us-east-1"      = "ami-0145beba9e1d0a7bc"
-    "us-west-2"      = "ami-0619a0df5948cd6db"
+    "ap-southeast-1" = "ami-02caaaa5ddda93489"
+    "ap-southeast-2" = "ami-0d40f660c5d10a298"
+    "eu-west-1"      = "ami-0df63dcc5da46143f"
+    "us-east-1"      = "ami-0f8dff21b5657007c"
+    "us-west-2"      = "ami-01bff1e42b5a675d2"
```
Comment (colinking) [inline main.tf:13]: Updated the AMIs.

**Pull request merged**